pkg/registry/apps/daemonset/storage/storage_test.go

```diff
@@ -193,4 +197,42 @@ func TestShortNames(t *testing.T) {
 	registrytest.AssertShortNames(t, storage, expected)
 }
 
-// TODO TestUpdateStatus
+func TestUpdateStatus(t *testing.T) {
+	storage, statusStorage, server := newStorage(t)
+	defer server.Terminate(t)
+	defer storage.Store.DestroyFunc()
+	ctx := genericapirequest.NewDefaultContext()
+
+	key, _ := storage.KeyFunc(ctx, "foo")
+	dsStart := newValidDaemonSet()
+	err := storage.Storage.Create(ctx, key, dsStart, nil, 0)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	dsIn := &extensions.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "foo",
+			Namespace: metav1.NamespaceDefault,
+		},
+		Status: extensions.DaemonSetStatus{
+			Conditions: []extensions.DaemonSetCondition{
+				{Status: "True"},
+			},
+		},
+	}
+
+	_, _, err = statusStorage.Update(ctx, dsIn.Name, rest.DefaultUpdatedObjectInfo(dsIn), rest.ValidateAllObjectFunc, rest.ValidateAllObjectUpdateFunc)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	obj, err := storage.Get(ctx, "foo", &metav1.GetOptions{})
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+	dsOut := obj.(*extensions.DaemonSet)
+	// only compare relevant changes b/c of difference in metadata
+	if !apiequality.Semantic.DeepEqual(dsIn.Status, dsOut.Status) {
```

Comment:
Maybe also make some changes to the spec and make sure it is not updated. Status update should not update spec.

Reply:
I think it is a good suggestion.